src/Microsoft.Extensions.DiagnosticAdapter/Internal/ProxyTypeEmitter.cs

```diff
@@ -101,8 +101,11 @@ private static bool VerifyProxySupport(ProxyBuilderContext context, Tuple<Type,
             }
 
             ProxyTypeCacheResult cacheResult;
+            var verificationResult = new VerificationResult();
             if (context.Cache.TryGetValue(key, out cacheResult))
             {
+                context.Visited.Add(key, verificationResult);
```

Comment:
Line 69 also need to change to a 'TryAdd`.

Basically we need to deal with the fact that we could find out from this method that our proxy type as already been created.